File: Fácil/Temperatures/Temperatures.lua
n = tonumber(io.read())
next_token = string.gmatch(io.read(), "[^%s]+")
res = 0
for i=0,n-1 do
    t = tonumber(next_token())
    if i == 0 then
        res = t
    end
    if math.abs(t) < math.abs(res) then
        res = t
    end
    if math.abs(t) == math.abs(res) then
        res = math.max(res, t)
    end
end
print(res)

File: Fácil/Temperatures/Temperatures.c
#include <stdlib.h>
#include <stdio.h>
#include <math.h>

int main()
{
    int n, res = 0;
    scanf("%d", &n);
    for (int i = 0; i < n; i++) {
        int t;
        scanf("%d", &t);
        res = !i ? t : res;
        res = abs(t) < abs(res) ? t : res;
        res = abs(t) == abs(res) ? fmax(t, res) : res;
    }
    printf("%d\n", res);
    return 0;
}
